### PR TITLE
Support snake case

### PR DESCRIPTION
**Pull request by adz**
Add snake case as per issue https://github.com/thoth-org/Thoth.Json/issues/47

Added Changelog which I hope is close to what you are after.

Let me know if you'd like anything changed.

### CODE REVIEW

File: docs/legacy/v2.md [hunk 1]
```diff
@@ -201,7 +201,7 @@ Auto decoder accept an optional argument `isCamelCase`:
 
 ```fsharp
 > let json = """{ "id" : 0, "name": "maxime", "email": "mail@domain.com", "followers": 0 }"""
-> Decode.Auto.fromString<User>(json, isCamelCase=true)
+> Decode.Auto.fromString<User>(json, caseStrategy=CamelCase)
```
Comment (MangelMaxime) [inline docs/legacy/v2.md]: This is the legacy documentation so we don't have to update it.

I will also need to "save" the v3 documentation without the new API change.

**Comment by MangelMaxime** [comment]
Thank you for the PR @adz, 

I fixed the documentation in order to remove the change you made to the legacy version of it.

I also ported the change to [Thoth.Json.Net](https://github.com/thoth-org/Thoth.Json.Net/tree/feature/snake_case).

So it's almost ready to release, I will do it on Friday as I don't have time for doing that before. I need to release new version of all the others library of Thoth ecosystem otherwise, the users will not be able to benefit from the new feature.

**Comment by adz** [comment]
Oh I didn't realize that was another copy... I thought the Thoth.Net lib
built on this one somehow.

I had a look at that now and noticed the paket dependencies seem to point
at my repo (adz) rather than the thoth.net one. I'm guessing that was
unintentionally done, so might need correction.

Thanks!

**Comment by MangelMaxime** [comment]
> I had a look at that now and noticed the paket dependencies seem to point
at my repo (adz) rather than the thoth.net one. I'm guessing that was
unintentionally done, so might need correction.

In the feature/snake_case this is normal because it's fetching the tests file via paket from Github. We are using the same tests file between Thoth.Json and Thoth.Json.Net to make sure both libraries encode/decode in the same way. It will be changed when merging this PR in master.

**Comment by MangelMaxime** [comment]
Thoth.Json 4.0.0 has been released :)

**Pull request merged**